ush/python/pygfs/task/chem_fire_emission.py

```diff
@@ -0,0 +1,949 @@
+            NRT_DIRECTORY = NRT_DIRECTORY.replace(self.start_date.strftime('%Y%m%d'),
+                                                  find_date.strftime('%Y%m%d'))
+            if not os.path.exists(NRT_DIRECTORY):
+                logger.warning(f"Directory does not exist: {NRT_DIRECTORY}")
+                continue
+            else:
+                break
+
+        if not os.path.exists(NRT_DIRECTORY):
+            logger.error(f"Could not find a valid NRT_DIRECTORY for GBBEPx files")
+            return []
+
+        all_files = os.listdir(NRT_DIRECTORY)
+        matching_files = []
+        logger.info(f"Searching in directory: {NRT_DIRECTORY}")
+        logger.debug(f"Total files in directory: {len(all_files)} files")
+        logger.debug(f"Files found in directory: {all_files}")
+        # Look for pattern: "GBBEPx-all01GRID_v4r0_blend_s202302240000000_e202302242359590_c202302250134090.nc"
+        pattern = r"GBBEPx-all01GRID.*_s(\d{8}).*_e(\d{8}).*\.nc"
+
+        for file_name in all_files:
+            match = re.match(pattern, file_name)
+            if match:
+                full_path = os.path.join(NRT_DIRECTORY, file_name)
+                matching_files.append(full_path)
+                logger.debug(f"Found GBBEPx NRT fire file: {full_path}")
+
+        # Remove duplicates while preserving order (safety check)
+        unique_files = []
+        seen = set()
+        for file_path in matching_files:
+            if file_path not in seen:
+                unique_files.append(file_path)
+                seen.add(file_path)
+
+        if len(unique_files) < len(matching_files):
+            logger.info(f"Found {len(unique_files)} unique GBBEPx NRT files (removed {len(matching_files) - len(unique_files)} duplicates)")
+
+        return unique_files
+
+    @logit(logger)
+    def _find_gbbepx_files(self, dates, aero_emis_fire_dir=None, version='v5r0'):
+        """Find GBBEPx files for the given date
+
+        Parameters
+        ----------
+        dates : str or list
+            Date or dates for which to find GBBEPx files
+        version : str
+            Version of GBBEPx files to search for
+
+        Returns
+        -------
+        List[str]
+            List of GBBEPx files for the given date(s)
+        """
+        logger.info(f'Finding GBBEPx files for {dates}')
+
+        # Find all possible months
+        months, years = self._get_unique_months()
+
+        # Format dates properly for matching
+        if not isinstance(dates, list):
+            dates = [dates]
+        date_strings = [d.strftime('%Y%m%d') if hasattr(d, 'strftime') else str(d) for d in dates]
+
+        files_found = []
+        # Find all possible files
+        for mon in months:
+            try:
+                emis_file_dir = aero_emis_fire_dir
+                if not os.path.exists(emis_file_dir):
+                    logger.warning(f"Directory does not exist: {emis_file_dir}")
+                    continue
```

Comment:
Since `aero_emis_fire_dir` is an input to this function, this will just continuously produce warnings for each month.  This should probably be moved above the `for` loop and cause a return of an empty list.